generators/generator-base.js

```diff
@@ -1756,9 +1756,14 @@ module.exports = class extends PrivateBase {
             context.fieldNamesUnderscored.push(_.snakeCase(field.fieldName));
             context.fieldNameChoices.push({ name: field.fieldName, value: field.fieldName });
         });
+        context.hasUserField = false;
         context.relationships.forEach((rel) => {
             context.relNameChoices.push({ name: `${rel.relationshipName}:${rel.relationshipType}`, value: `${rel.relationshipName}:${rel.relationshipType}` });
+            if (rel.relationshipName === 'user') {
+                context.hasUserField = true;
+            }
         });
+        context.saveIdentitySnapshot = context.applicationType === 'microservice' && context.authenticationType === 'oauth2' && context.hasUserField;
```

Comment:
I didnt take a  close look but may be this should be done in the loadInMemoryData section, as this method might not be called when creating entity from CLI. There is a forEach loop on relations there as well.

Reply:
@mraible could you let me know if that works?

Reply:
It does work - thanks @deepu105! 🙏